vault/expiration.go

```diff
@@ -319,6 +364,19 @@ func (m *ExpirationManager) Restore(errorFunc func()) (retErr error) {
 		}
 	}()
 
+	if atomic.LoadUint32(&m.SHA1HashedUnleasedTokensUpgraded) == 0 {
```

Comment:
Same here, i think this should be checking `SHA1HashedUnleasedTokensUpgraded` insead